tools/scripts/helpers/issue.py

```diff
@@ -0,0 +1,162 @@
+    "type-unsupported": "unsupported_message"
+}
+
+TRACKED_MILESTONES = [
+    "needsdiagnosis",
+    "duplicate",
+    "moved",
+    "contactready",
+    "needscontact",
+    "sitewait"
+]
+
+
+@dataclass
+class WebcompatIssue:
+    body: str
+    breakage_url: str
+    number: int
+    title: str
+    labels: List[str]
+    milestone: str
+    timeline: List[dict]
+
+    @classmethod
+    def from_dict(cls, issue: dict):
+        original_labels = issue.get("labels", [])
+        labels = [label["name"] for label in original_labels]
+        issue_body = issue["body"]
+        breakage_url = extract_url(issue_body)
+
+        milestone = ""
+        if issue.get("milestone"):
+            milestone = issue["milestone"]["title"]
+
+        return cls(body=issue_body,
+                   breakage_url=breakage_url,
+                   number=issue["number"],
+                   title=issue["title"],
+                   labels=labels,
+                   milestone=milestone,
+                   timeline=issue["timeline"])
+
+    def build_canonical_url(self) -> str:
+        return WEBCOMPAT_URL + str(self.number)
+
+    def get_platform(self) -> list:
+        browser = []
+        for label in self.labels:
+            if label in BROWSER_MAP:
+                browser.append(BROWSER_MAP[label])
+
+        if len(browser) >= 2:
+            browser = ['all']
+
+        return browser
+
+    def get_last_reproduced(self) -> str:
+        if self.milestone in TRACKED_MILESTONES:
+            for timeline_item in self.timeline:
+                if (timeline_item["event"] == "milestoned"
+                        and timeline_item["milestone"]["title"] == self.milestone):
+                    return to_datetime(timeline_item["created_at"]).strftime('%Y-%m-%d')
+
+        return ""
+
+    def estimate_impact(self) -> str:
+        if "type-unsupported" in self.labels:
+            return SEVERITY_MAP["type-unsupported"]
+
+        severity_label = [label for label in self.labels if label.startswith('severity-')]
+
+        if severity_label:
+            return SEVERITY_MAP[severity_label[0]]
+
+        return "Can't estimate impact, please enter it manually."
+
+    def generate_report(self) -> dict:
+        is_fixed = False
+
+        report = {
+            "url": self.build_canonical_url(),
+            "site": self.breakage_url,
+            "platform": self.get_platform()
+        }
+
+        if self.milestone in RESOLVED_MAP.keys():
+            is_fixed = True
+            report["resolution"] = RESOLVED_MAP[self.milestone]
+
+        if "sitepatch-applied" in self.labels:
+            is_fixed = True
+            report["intervention"] = "Intervention is detected for this issue. Please locate it manually."
+            report["resolution"] = RESOLVED_MAP["intervention"]
+
+        if is_fixed:
+            report["impact"] = "minor_visual"
+            report["affects_users"] = "few"
```

Comment:
Not sure about this, but picked the lowest possible `impact` and `affects_users` for issues that are fixed / site changed or have intervention. We could check each fixed issue manually for these two parameters, but wonder if there is value in that?

Reply:
I don't think we want to do this because it means that we're double-counting the fact that an issue was fixed. If a particular platform issue has a history of leading to site-breaking problems, we shoud prioritise that higher than one which has a history of causing minor visual issues, even if all the cases we know about are fixed. So I think we want the original impact/affects_users values here. Initially that's more work, but if we start adding issues before they're fixed it will be less work to just update the status.

Reply:
Agree, that's a good point :)